modules/engines/engine-odata/src/main/java/org/eclipse/dirigible/engine/odata2/handler/ScriptingOData2EventHandler.java

```diff
@@ -39,346 +39,351 @@
-	private static final String DIRIGIBLE_ODATA_HANDLER_EXECUTOR_ON_EVENT = "DIRIGIBLE_ODATA_HANDLER_EXECUTOR_ON_EVENT";
 
-	private static final Logger logger = LoggerFactory.getLogger(DefaultSQLProcessor.class);
-	
-	private static final String DIRIGIBLE_ODATA_WRAPPER_MODULE_ON_EVENT = "odata/wrappers/onEvent";
+    private static final String DIRIGIBLE_ODATA_HANDLER_EXECUTOR_TYPE = "DIRIGIBLE_ODATA_HANDLER_EXECUTOR_TYPE";
+    private static final String DIRIGIBLE_ODATA_HANDLER_EXECUTOR_ON_EVENT = "DIRIGIBLE_ODATA_HANDLER_EXECUTOR_ON_EVENT";
 
-	private static final String ERROR_EXECUTING_XSJSLIB_HANDLER = "Error executing .xsjslib handler: ";
-	
-	private ODataCoreService odataCoreService = new ODataCoreService();
+    private static final Logger logger = LoggerFactory.getLogger(DefaultSQLProcessor.class);
 
-	@Override
-	public void beforeCreateEntity(PostUriInfo uriInfo, String requestContentType,
-			String contentType, ODataEntry entry, Map<Object, Object> context) throws ODataException {
-		try {
-			String namespace = uriInfo.getTargetType().getNamespace();
-			String name = uriInfo.getTargetType().getName();
-			String method = ODataHandlerMethods.create.name();
-			String type = ODataHandlerTypes.before.name();
-			List<ODataHandlerDefinition> handlers = odataCoreService.getHandlers(namespace, name, method, type);
-			context.put("uriInfo", uriInfo);
-			context.put("requestContentType", requestContentType);
-			context.put("contentType", contentType);
-			context.put("entry", entry);
-			executeHandlers(handlers, context);
-		} catch (EdmException | ODataException e) {
-			logger.error(e.getMessage(), e);
-		} catch (ScriptingException e) {
-			throw new ODataException(ERROR_EXECUTING_XSJSLIB_HANDLER + e.getMessage(), e);
-		}
-	}
+    private static final String DIRIGIBLE_ODATA_WRAPPER_MODULE_ON_EVENT = "odata/wrappers/onEvent";
 
-	@Override
-	public void afterCreateEntity(PostUriInfo uriInfo, String requestContentType,
-			String contentType, ODataEntry entry, Map<Object, Object> context) throws ODataException {
-		try {
-			String namespace = uriInfo.getTargetType().getNamespace();
-			String name = uriInfo.getTargetType().getName();
-			String method = ODataHandlerMethods.create.name();
-			String type = ODataHandlerTypes.after.name();
-			List<ODataHandlerDefinition> handlers = odataCoreService.getHandlers(namespace, name, method, type);
-			context.put("uriInfo", uriInfo);
-			context.put("requestContentType", requestContentType);
-			context.put("contentType", contentType);
-			context.put("entry", entry);
-			executeHandlers(handlers, context);
-		} catch (EdmException | ODataException e) {
-			logger.error(e.getMessage(), e);
-		} catch (ScriptingException e) {
-			throw new ODataException(ERROR_EXECUTING_XSJSLIB_HANDLER + e.getMessage(), e);
-		}
-	}
+    private static final String ERROR_EXECUTING_XSJSLIB_HANDLER = "Error executing .xsjslib handler: ";
```

Comment:
This is generic scripting handler for the dirigible, it is not only for xsjslib functions. It is also used with the standard Dirigible OData events where the handler function could be in a js file. This should be a generic message.